### PR TITLE
Added functionality to prompt update for version ,run and add commands when required #388

### LINKED ISSUE #388: Prompt upgrade when CLI detects new version

### PR DESCRIPTION
**Pull request by Adm28**
Invoked GetLatestCliRelease() to check if the current installed CLI version matches with the latest release's version.
If not user is prompted to run the upgrade command
> spice version
```
CLI version:     local
Runtime version: not installed
Note: New CLI version v0.3.1-alpha is now available!
Note: Run "spice upgrade" to update CLI 
```

Closes #388

### CODE REVIEW

File: pkg/cli/cmd/run.go [hunk 1]
```diff
@@ -17,10 +19,22 @@ spice run
 `,
 	Run: func(cmd *cobra.Command, args []string) {
 		err := runtime.Run(contextFlag, "")
```
Comment (lukekim) [inline pkg/cli/cmd/run.go]: Need to check `err`

File: pkg/cli/cmd/add.go [hunk 2]
```diff
@@ -35,6 +37,16 @@ spice add samples/LogPruner
 		relativePath := context.CurrentContext().GetSpiceAppRelativePath(downloadPath)
 
 		cmd.Printf("Added %s\n", relativePath)
+
+		release, err := github.GetLatestCliRelease()
```
Comment (lukekim) [inline pkg/cli/cmd/add.go]: Instead of this block three times, probably best to refactor into something like `pkg/cli/version.go`

**Review by lukekim — CHANGES_REQUESTED** [review]

File: pkg/cli/cmd/add.go (hunk 2 above)
Comment (Adm28) [reply pkg/cli/cmd/add.go]: So if I understand this block needs to be wrapped into a function and exposed, correct me if my understanding is wrong here

File: pkg/cli/cmd/run.go (hunk 1 above)
Comment (Adm28) [reply pkg/cli/cmd/run.go]: Noted

File: pkg/cli/cmd/add.go (hunk 2 above)
Comment (lukekim) [reply pkg/cli/cmd/add.go]: That would work, but you can make it a private package function (lowercase function name) so it's only available where you use it.

File: pkg/github/runtime_release.go [hunk 3]
```diff
@@ -17,6 +18,18 @@ const (
 	runtimeRepo  = "spiceai"
 )
 
+func CheckForLatestVersion() {
```
Comment (lukekim) [inline pkg/github/runtime_release.go]: As this is CLI specific it should probably go in something like `pkg/cli/version.go`

Comment (lukekim) [reply pkg/github/runtime_release.go]: It should also return the error, which should be checked by the caller. I.e.

```go
func checkForLatestVersion() error {
```

File: pkg/cli/cmd/run.go (hunk 1 above)
Comment (lukekim) [reply pkg/cli/cmd/run.go]: Probably want to keep the error print and os.Exit(1).

**Review by lukekim — APPROVED** [review]

File: pkg/github/runtime_release.go (hunk 3 above)
Comment (Adm28) [reply pkg/github/runtime_release.go]: Got your point. Will do the changes now

**Review by phillipleblanc — APPROVED** [review]

File: pkg/cli/cmd/upgrade.go [hunk 4]
```diff
@@ -18,6 +19,7 @@ var upgradeCmd = &cobra.Command{
 spice upgrade
 `,
 	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("Checking for latest Spice CLI release...")
```
Comment (lukekim) [inline pkg/cli/cmd/upgrade.go]:
```suggestion
		cmd.Println("Checking for latest Spice CLI release...")
```

File: pkg/cli/cmd/version.go [hunk 5]
```diff
@@ -43,9 +45,27 @@ spice version
...
 }
 
+func checkLatestCliReleaseVersion() error {
+	release, err := github.GetLatestCliRelease()
+	if err != nil {
+		return err
+	}
+	cliVersion := version.Version()
+	if cliVersion != release.TagName {
+		fmt.Printf("Note: New CLI version %s is now available!\nNote: Run \"spice upgrade\" to update CLI \n", release.TagName)
```
Comment (lukekim) [inline pkg/cli/cmd/version.go]:
```suggestion
		fmt.Printf("New CLI version %s is now available!\nRun \"spice upgrade\" to upgrade.\n", release.TagName)
```

**Review by lukekim — APPROVED** [review]

**Review by lukekim — APPROVED** [review]
Thanks!

**Pull request merged**